### PR TITLE
Feat: Improve variable listener event query

### PR DESCRIPTION
**Pull request by canchito-dev**
Trigger querying the database table ACT_RU_EVENT_SUBSCR only if the variable that is created/updated has a variable listener event associated with it.
The change only applies to processes.

Check List:
- Unit tests: Yes. Those already available
- Documentation: No change here

### CODE REVIEW

File: modules/flowable-eventsubscription-service/src/main/java/org/flowable/eventsubscription/service/impl/persistence/entity/data/impl/MybatisEventSubscriptionDataManager.java [hunk 1]
```diff
@@ -12,11 +12,7 @@
  */
 package org.flowable.eventsubscription.service.impl.persistence.entity.data.impl;
 
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (dbmalkovsky) [inline modules/flowable-eventsubscription-service/src/main/java/org/flowable/eventsubscription/service/impl/persistence/entity/data/impl/MybatisEventSubscriptionDataManager.java:15]: same note as above; thanks

File: modules/flowable-eventsubscription-service/src/main/java/org/flowable/eventsubscription/service/impl/persistence/entity/data/impl/MybatisEventSubscriptionDataManager.java [hunk 2]
```diff
@@ -37,23 +33,7 @@
...
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.EventSubscriptionsBySubScopeIdMatcher;
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.MessageEventSubscriptionsByProcInstAndEventNameMatcher;
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.SignalEventSubscriptionByEventNameMatcher;
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.SignalEventSubscriptionByNameAndExecutionMatcher;
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.SignalEventSubscriptionByProcInstAndEventNameMatcher;
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.SignalEventSubscriptionByScopeAndEventNameMatcher;
-import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.SignalEventSubscriptionByScopeIdAndTypeMatcher;
+import org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher.*;
```
Comment (dbmalkovsky) [inline modules/flowable-eventsubscription-service/src/main/java/org/flowable/eventsubscription/service/impl/persistence/entity/data/impl/MybatisEventSubscriptionDataManager.java:36]: and again.

File: modules/flowable-eventsubscription-service/src/main/java/org/flowable/eventsubscription/service/impl/persistence/entity/data/impl/cachematcher/EventSubscriptionsByProcInstTypeAndNameMatcher.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flowable.eventsubscription.service.impl.persistence.entity.data.impl.cachematcher;
+
+import org.flowable.common.engine.impl.persistence.cache.CachedEntityMatcherAdapter;
+import org.flowable.eventsubscription.service.impl.persistence.entity.EventSubscriptionEntity;
+
+import java.util.Map;
+import java.util.Set;
```
[the remark this answers is not in the document]
Comment (canchito-dev) [reply modules/flowable-eventsubscription-service/src/main/java/org/flowable/eventsubscription/service/impl/persistence/entity/data/impl/cachematcher/EventSubscriptionsByProcInstTypeAndNameMatcher.java:16]: Idea settings applied ;)

**Comment by filiphr** [comment]
@canchito-dev what kind of a problem are you trying to solve with this PR?

**Comment by canchito-dev** [comment]
Hi @filiphr
The main objective is to reduce the amount of select queries to the database when there is a variable listener, by only executing it, if the variable that is being created/updated has an associated active listener.

**Comment by tijsrademakers** [comment]
Hi @canchito-dev ,

Thank you for providing the PR, we have made some modifications and did the commit in https://github.com/flowable/flowable-engine/commit/06be453838f88a61e6d72ba5378d236504167e86

Best regards

**Pull request closed without merging**